src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -187,6 +208,19 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
     const std::string element_name =
         MakeString{} << ElementId<VolumeDim>(array_index) << '/';
 
+    // calculate the interpolation matrices if requested. Else, interp_matrices
+    // will not be default constructed and therefore skipped by applyMatrices.
+    std::array<Matrix, VolumeDim> interp_matrices{};
+    if (interpolation_mesh_) {
+      const auto slices = mesh.slices();
+      const auto interpolation_slices = interpolation_mesh_->slices();
+      for (size_t i = 0; i < VolumeDim; i++) {
```

Comment:
Prefer `++i` over `i++`. It's good to get in the habit of prefix-incrementing to avoid copies in cases where they actually matter.